src/interface/state_descriptor.hpp

```diff
@@ -67,10 +70,17 @@ class StateDescriptor {
   // infers type correctly.
   template <typename T>
   const T &Param(const std::string &key, T value) {
-    params_.Get(key, value);
+    return params_.Get(key, value);
+  }
+
+  const std::type_index &ParamType(const std::string &key) {
+    return params_.GetType(key);
   }
 
   Params &AllParams() { return params_; }
+
+  std::vector<std::string> &IncludeWithOutputParams() { return includeWithOutputParams_; }
+
```

Comment:
Can be made const noexcept.

Reply:
It looks like you are also only adding to includeWithOutputParams_ from within the Statedescriptor class so you could probably return by const reference as well.